src/DMXSerial2.cpp

```diff
@@ -829,7 +861,131 @@ void DMXSerialClass2::_processRDMMessage(byte CmdClass, uint16_t Parameter, bool
+            handled = true;
+            ///////////////////////////////////////////
+          }
+        }
+      } else if (CmdClass == E120_SET_COMMAND) {
+        // Unexpected set
+        nackReason = E120_NR_UNSUPPORTED_COMMAND_CLASS;
+      }
+
+    } else if (Parameter == SWAPINT(E120_SENSOR_VALUE) && _initData->sensorsLength > 0) { // 0x0201
+      if (CmdClass == E120_GET_COMMAND) {
+        if (_rdm.packet.DataLength != 1) {
+          // Unexpected data
+          nackReason = E120_NR_FORMAT_ERROR;
+        } else if (_rdm.packet.SubDev != 0) {
+          // No sub-devices supported
+          nackReason = E120_NR_SUB_DEVICE_OUT_OF_RANGE;
+        } else {
+          uint8_t sensorNr = _rdm.packet.Data[0];
+          if (sensorNr >= _initData->sensorsLength || sensorNr == 0xFF) {
+            // Out of range sensor
+            nackReason = E120_NR_DATA_OUT_OF_RANGE;
+          } else {
+            int16_t sensorValue = 0;
+            int16_t lowestValue = 0;
+            int16_t highestValue = 0;
+            int16_t recordedValue = 0;
+            bool8 res = false;
+            if (_sensorFunc) {
+              res = _sensorFunc(sensorNr, &sensorValue, &lowestValue, &highestValue, &recordedValue);
+            }
+            if (res) {
+              _rdm.packet.DataLength = 9;
+              _rdm.packet.Data[0] = sensorNr;
+              WRITEINT(_rdm.packet.Data +  1, sensorValue);
+              WRITEINT(_rdm.packet.Data +  3, (_initData->sensors[sensorNr].lowHighSupported ? lowestValue : 0));
+              WRITEINT(_rdm.packet.Data +  5, (_initData->sensors[sensorNr].lowHighSupported ? highestValue : 0));
+              WRITEINT(_rdm.packet.Data +  7, (_initData->sensors[sensorNr].recordedSupported ? recordedValue : 0));
+              handled = true;
+            } else {
+              nackReason = E120_NR_HARDWARE_FAULT;
+            }
+          }
+        }
+      } else if (CmdClass == E120_SET_COMMAND) {
+        // Set on a sensor is used to reset stats.
+        if (_rdm.packet.DataLength != 1) {
+          // Unexpected data
+          nackReason = E120_NR_FORMAT_ERROR;
+        } else if (_rdm.packet.SubDev != 0) {
+          // No sub-devices supported
+          nackReason = E120_NR_SUB_DEVICE_OUT_OF_RANGE;
+        } else {
+          uint8_t sensorNr = _rdm.packet.Data[0];
+          if (sensorNr >= _initData->sensorsLength && sensorNr != 0xFF) {
+            // Out of range sensor
+            nackReason = E120_NR_DATA_OUT_OF_RANGE;
+          } else {
+            for (int i = 0; i<_initData->sensorsLength; i++) {
+              if (i == sensorNr || sensorNr == 0xFF) {
+                int16_t sensorValue = 0;
+                int16_t lowestValue = 0;
+                int16_t highestValue = 0;
+                int16_t recordedValue = 0;
+                bool8 res = false;
+                if (_sensorFunc) {
+                  res = _sensorFunc(i, &sensorValue, &lowestValue, &highestValue, &recordedValue);
```

Comment:
We need to do something clever here if we're in 0xff mode. Ideally ANDing all the results, but obviously not the default initialisation...

Tracking failure would mean we could OR it I think.